### ISSUE TITLE
Opam does not respect the opam-root configuration

### ISSUE THREAD

**Issue by silene** [ocaml/opam#3668]
The `~/.opam/4.06.1/.opam-switch/switch-config` file contains the line:

    opam-root: "/home/.ecryptfs/guillaume/.opam"

I would expect opam to use this directory as the root directory. Unfortunately, that is not the case, since opam uses `~/.opam` as the root. I have to set the `OPAMROOT` environment variable to force opam to use the correct root. Am I misunderstanding what the `opam-root` configuration setting does?

Note that this only matters because of sandboxing. Without sandboxing, whichever root opam uses is fine, as both directories are ultimately the same.

```
# opam config report
# opam-version      2.0.1 
# self-upgrade      no
# system            arch=x86_64 os=linux os-distribution=ubuntu os-version=18.04
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              3
# repositories      1 (http), 1 (version-controlled) (default repo at 9bea3748)
# pinned            0
# current-switch    4.06.1
```

**Comment by rjbou** [comment]
The [`opam-root`](https://opam.ocaml.org/doc/Manual.html#switchconfigfield-opam-root) field is used for local switches, to know to which opam root the switch belongs to.
Why do you need to switch opam root on sanbdoxed operations?

**Comment by silene** [comment]
Otherwise, builds fail with failures such as `Can't mkdir /home/guillaume/.opam/4.06.1: No such file or directory`. This is the same kind of setup as #3622 . I have not yet tested if #3661 would alleviate the issue.

**Comment by rjbou** [comment]
It seems more related to #3622, opam doesn't resolve the default opam root (cf. https://github.com/ocaml/opam/issues/3622#issuecomment-441029473 & #3681).
Is your `~/.opam` a symlink to `/home/.ecryptfs/guillaume/.opam`?

**Comment by silene** [comment]
> Is your `~/.opam` a symlink to `/home/.ecryptfs/guillaume/.opam`?

Yes. Sorry that wasn't clear.

**Comment by rjbou** [comment]
Fix #3681 merged on master & will be available on 2.0.2 release.
Thanks for reporting!
